include/dem/dem_action_manager.h

```diff
@@ -0,0 +1,429 @@
+#ifndef dem_action_manager_h
+#define dem_action_manager_h
+
+class DEMActionManager
+{
+public:
+  /**
+   * @brief Copy constructor as a delete function to make sure it can not be
+   * copied. Will never be used.
+   *
+   * @param copy The object to be copied
+   */
+  DEMActionManager(const DEMActionManager &copy) = delete;
+
+  /**
+   * @brief Copy constructor as a delete function to make sure it can not be
+   * assigned. Will never be used.
+   *
+   * @param copy The object to be copied
+   */
+  DEMActionManager &
+  operator=(const DEMActionManager &copy) = delete;
+
+  /**
+   * @brief Getter of the unique instance of the DEMActionManager.
+   */
+  static DEMActionManager *
+  get_action_manager();
+
+  /**
+   * @brief Reset all triggers to false.
+   * TODO
```

Comment:
TODO here being setting the documentation?

Reply:
I want to remove the `contact_search_trigger = mobility_status_reset_trigger;` because it's sketchy and not clear haha